packer/setup.rb

```diff
@@ -27,21 +33,24 @@ class SystemPackerBuildError < RuntimeError; end
 
 # Checked system - error if the command fails
 def csystem(cmd, err, opts = {})
-  out = `#{cmd}`
-  print "Running command: #{cmd.inspect}\n" if opts[:debug] || opts["debug"]
+  print "Running command: #{cmd.inspect}\n" if VERBOSE || opts[:debug] || opts["debug"]
+  if VERBOSE
+    system(cmd)
+  else
+    out = `#{cmd}`
+  end
   unless $?.success? || opts[:fail_ok] || opts["fail_ok"]
-    print "Error running command:\n#{cmd.inspect}\nOutput:\n#{out}\n=====\n"
+    puts "Error running command:\n#{cmd.inspect}"
+    puts "Output:\n#{out}\n=====" if out
     raise SystemPackerBuildError.new(err)
   end
-  print "Command output:\n#{out}\n=====\n" if opts[:debug] || opts["debug"]
-  out
 end
 
 def clone_or_update_repo(repo_url, tag, work_dir)
   tag = tag.strip
   if File.exist?(work_dir)
     Dir.chdir(work_dir) do
-      csystem "git pull", "Couldn't 'git pull' in #{work_dir}!"
+      csystem "git fetch", "Couldn't 'git fetch' in #{work_dir}!"
```

Comment:
I was thinking we checkout the tag from `benchmark_software.json`, so therefore we only need to fetch + checkout TAG. I'll adapt the logic to either checkout the tag or pull if there is no tag.